src/LanguageServer.php

```diff
@@ -100,4 +120,42 @@ public function exit()
     {
         exit(0);
     }
+
+    /**
+     * Parses workspace files, one at a time.
+     *
+     * @param string $rootPath The rootPath of the workspace. Is null if no folder is open.
+     * @return void
+     */
+    private function indexProject(string $rootPath)
+    {
+        $dir = new \RecursiveDirectoryIterator($rootPath);
+        $ite = new \RecursiveIteratorIterator($dir);
+        $files = new \RegexIterator($ite, '/^.+\.php$/i', \RegexIterator::GET_MATCH);
+        $fileList = array();
+        foreach($files as $file) {
+            $fileList = array_merge($fileList, $file);
+        }
+        
+        $processFile = function() use (&$fileList, &$processFile, &$rootPath){
+            if ($file = array_pop($fileList)) {
+                
+                $uri = 'file://'.(substr($file, -1) == '/' || substr($file, -1) == '\\' ? '' : '/').str_replace('\\', '/', $file);
+                
+                $numFiles = count($fileList);
+                if (($numFiles % 100) == 0) {
+                    $this->client->window->logMessage(3, $numFiles.' PHP files remaining.');
+                }
+                
+                $this->project->getDocument($uri)->updateAst(file_get_contents($file));
+                
+                Loop\nextTick($processFile);
```

Comment:
I think this should be `setTimeout`. `nextTick` will queue the callback before any I/O callbacks. `setTimeout` on the other hand will push it to the end of the queue, first executing all other callbacks (like the STDIN handling for example), then getting back to processing. See this note from the Node docs:

> This is not a simple alias to setTimeout(fn, 0), it's much more efficient. It runs before any additional I/O events (including timers) fire in subsequent ticks of the event loop.
> Note: the next tick queue is completely drained on each pass of the event loop before additional I/O is processed. As a result, recursively setting nextTick callbacks will block any I/O from happening, just like a while(true); loop.

The current way would mean that I/O is blocked (including STDIN), so the language server would become unresponsive until the indexing is finished.